library/globals.inc.php

```diff
@@ -2976,7 +2976,7 @@ function gblTimeZones()
         'site_addr_oath' => array(
             xl('Site Address (required for OAuth2, FHIR and CCDA)'),
             'text',
-            'https://' . $_SERVER['HTTP_HOST'] . $GLOBALS['webroot'],
+            $ResolveServerHost(), // anonymous function in globals.php
```

Comment:
Another option here is making this blank by default and changing the wording of the setting to be an override in the case that the automatic method does not work. Then the automatic method will get much more attention over the longer term (since will be used on every globals.php call) and can be adjusted (in addition to correctly configuring apache/nginx settings) as it gets exposed to more use cases.
Maybe something like
quick desc - Site Address Manual Override
long desc - Site Address Manual Override. Only need to set this if the automatic method does not work for OAuth. An example setting would be https://localhost:8300 .

Reply:
okay. that may be less confusing.